packages/web-frontend/src/utils/measures.js

```diff
@@ -204,6 +205,16 @@ export function processMeasureInfo(response) {
     };
   });
 
+  // remove measure units with no coordinates
+  // for circle heatmap remove empty values or values that are not of positive float type
+  if (displayType === MARKER_TYPES.CIRCLE_HEATMAP) {
+    measureData = measureData.filter(({ value }) => {
+      if (!value || value === '') return false;
+      const parsedValue = parseFloat(value);
+      return !Number.isNaN(parsedValue) && parsedValue >= 0;
+    });
+  }
+
```

Comment:
This was from `mapReducer.js`, moving to `processMeasureInfo` makes more sense to me